### PR TITLE
Add alert Info in Map

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- #6 

## Changes
- Add map alert details 

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -3,20 +3,34 @@ import 'mapbox-gl/dist/mapbox-gl.css';
...
+    ApolloProvider,
+    InMemoryCache,
+} from '@apollo/client';
 
 import App from './App';
 
 const webappRootId = 'webapp-root';
 
 const webappRootElement = document.getElementById(webappRootId);
+const APP_GRAPHQL_ENDPOINT = 'http://localhost:8000/graphql/';
```
Comment (samshara) [inline src/index.tsx]: Let's get this from `env` variables

File: src/utils/constants.ts [hunk 2]
```diff
@@ -3,7 +3,11 @@ export const DURATION_MAP_ZOOM = 1000;
 export const DEFAULT_MAP_PADDING = 50;
 
 // Colors
-
+export const COLOR_LIGHT_GREY = '#e0e0e0';
+export const COLOR_RED = '#f5333f';
+export const COLOR_PRIMARY_RED = '#f5333f';
```
Comment (samshara) [inline src/utils/constants.ts]: Why use different variable name for the same color?

File: src/views/AlertMap/index.tsx [hunk 3]
```diff
@@ -1,44 +1,252 @@
...
+                parameter
+                parameters {
+                  id
+                }
+              }
+              countryId
+            }
+            limit
+            offset
+        }
```
Comment (samshara) [inline src/views/AlertMap/index.tsx]: Do we require all these fields?

File: vite.config.ts [hunk 4]
```diff
@@ -14,6 +14,8 @@ const commitHash = execSync('git rev-parse --short HEAD').toString();
 
 export default defineConfig(({ mode }) => {
     const isProd = mode === 'production';
+    const env = loadEnv(mode, process.cwd(), '')
+
```
Comment (samshara) [inline vite.config.ts:17]: variable defined but seems to be unused

File: src/views/AlertMap/index.tsx [hunk 5]
```diff
@@ -51,20 +259,68 @@ function OngoingAlertMap(props: Props) {
...
+                    </MapPopup>
+                )}
                 <MapBounds
                     duration={DURATION_MAP_ZOOM}
                     bounds={bbox}
                     padding={DEFAULT_MAP_PADDING}
                 />
             </BaseMap>
+            <Container
+                className={styles.countryList}
```
Comment (samshara) [inline src/views/AlertMap/index.tsx]: this style definition is re-used, was this intentional?

File: docker-compose.yml [hunk 6]
```diff
@@ -7,3 +7,5 @@ services:
       - .:/code
     ports:
       - '3000:3000'
+        environment:
+ APP_GRAPHQL_ENDPOINT: ${APP_GRAPHQL_ENDPOINT:-http://localhost:8000/graphql/}
```
Comment (tnagorra) [inline docker-compose.yml]: Is this correct?

File: package.json [hunk 7]
```diff
@@ -15,20 +15,26 @@
...
         "@graphql-codegen/introspection": "^4.0.3",
+        "@graphql-codegen/typescript": "^4.0.6",
         "@graphql-codegen/typescript-operations": "^4.2.0",
         "@ifrc-go/icons": "^1.3.3",
         "@ifrc-go/ui": "^1.0.0",
         "@mapbox/mapbox-gl-draw": "^1.4.3",
-        "@sentry/react": "^7.81.1",
         "@togglecorp/fujs": "^2.1.1",
         "@togglecorp/re-map": "^0.2.0-beta-6",
+        "@togglecorp/toggle-form": "^2.0.4",
```
Comment (tnagorra) [inline package.json]: Do we need form library?

File: vite.config.ts (hunk 4 above)
Comment (tnagorra) [reply vite.config.ts:17]: @samshara Why did we resolve this?

File: src/views/Home/AlertsView/index.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+import AlertsAside from './AlertsAside';
+import AlertsMap from './AlertsMap';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const COUNTRIES_LIST = gql`
+query CountryList {
+  public {
+    allCountries(alertFilters: {}) {
```
Comment (tnagorra) [inline src/views/Home/AlertsView/index.tsx:26]:
```suggestion
    allCountries {
```

Can we not skip the filters altogether?

File: src/views/Home/AlertsView/index.tsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    const {
+        data: countryResponse,
+        loading: countryLoading,
+    } = useQuery<CountryListQuery, CountryListQueryVariables>(
+        COUNTRIES_LIST,
+    );
+
+    const countriesWithAlert = countryResponse?.public.allCountries.filter(
+        (country) => (country?.filteredAlertCount ?? 0) > 0,
+    );
```
Comment (tnagorra) [inline src/views/Home/AlertsView/index.tsx]: Let's memoize this

File: src/views/Home/AlertsView/index.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+            <AlertsMap
+                className={styles.alertsMap}
+                countriesWithAlert={countriesWithAlert}
+            />
+            <AlertsAside
+                className={styles.alertsAside}
+                countriesWithAlert={countriesWithAlert}
+                alertsFiltered={false} // NOTE: set this when the data is filtered
+                alertsPending={countryLoading}
+                alertsFetchError={false} // NOTE: set this on error
```
Comment (tnagorra) [inline src/views/Home/AlertsView/index.tsx]: Let's pass error

File: src/views/Home/AlertsView/AlertsAside/CountryListItem/index.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import {
+    Button,
+    Container,
+} from '@ifrc-go/ui';
+
+import { CountryListQuery } from '#generated/types/graphql';
+
+import styles from './styles.module.css';
+
+type CountryType = NonNullable<NonNullable<CountryListQuery['public']>['allCountries']>[number];
```
Comment (tnagorra) [inline src/views/Home/AlertsView/AlertsAside/CountryListItem/index.tsx]:
```suggestion
type Country = NonNullable<NonNullable<CountryListQuery['public']>['allCountries']>[number];
```

File: src/utils/map.ts [hunk 12]
```diff
@@ -17,16 +17,16 @@ export const defaultNavControlPosition: ControlPosition = 'top-right';
...
+    minZoom: 0,
     maxZoom: 18,
     scrollZoom: false,
     pitchWithRotate: false,
     dragRotate: false,
     renderWorldCopies: true,
     attributionControl: false,
     preserveDrawingBuffer: true,
-    // interactive: false,
+    center: [34, 39],
```
Comment (tnagorra) [inline src/utils/map.ts:30]: Let's add a NOTE why we are using this center

File: docker-compose.yml [hunk 13]
```diff
@@ -7,3 +7,10 @@ services:
       - .:/code
     ports:
       - '3000:3000'
+    environment:
+      APP_TITLE: ${APP_TITLE:-IFRC Alert Hub}
+      APP_ENVIRONMENT: ${APP_ENVIRONMENT:-testing}
+      APP_GRAPHQL_API_ENDPOINT: ${APP_GRAPHQL_API_ENDPOINT:-http://localhost:8000/graphql/}
+      APP_GRAPHQL_CODEGEN_ENDPOINT: ${APP_GRAPHQL_CODEGEN_ENDPOINT:-http://localhost:8000/graphql/}
+      APP_MAPBOX_ACCESS_TOKEN: ${APP_MAPBOX_ACCESS_TOKEN}
```
Comment (tnagorra) [inline docker-compose.yml]: @thenav56 Do we need this line?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Home/AlertsView/index.tsx (hunk 10 above)
Comment (barshathakuri) [reply src/views/Home/AlertsView/index.tsx]: For apollo client we are using boolean value

**Review by puranban — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**